File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.py
def solution(s, skip, index):
    alphabet_str = 'abcdefghijklmnopqrstuvwxyz';
    alphabet_lst = list(alphabet_str)
    for i in skip:
        alphabet_lst.remove(i)
        
    cur_alphabet_length = len(alphabet_lst) 
    
    result = ''
    
    for alphabet in s:
        cur_idx = alphabet_lst.index(alphabet) + index
        if cur_idx < cur_alphabet_length:
            result += alphabet_lst[cur_idx]
        else:
            cur_idx = cur_idx % cur_alphabet_length
            result += alphabet_lst[cur_idx]
    return result

File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.js
const solution = (s, skip, index) => {
    let alphabet_str = 'abcdefghijklmnopqrstuvwxyz';
    let alphabet_lst = alphabet_str.split('').filter((a) => !skip.includes(a));
    
    const alphabet_len = alphabet_lst.length;
    
    let result = '';
    
    s.split('').forEach((alphabet) => {
        let cur_index = alphabet_lst.indexOf(alphabet) + index;
        if (cur_index < alphabet_len) {
            result += alphabet_lst[cur_index];
        } else {
            result += alphabet_lst[cur_index % alphabet_len];
        }
    });
    return result;
}